Replace null_resource with terraform_data

janfrederik opened the pull request:
As from Terraform 1.4, the doc suggests to use `terraform_data` instead of `null_resource`. See https://registry.terraform.io/providers/hashicorp/null/latest/docs/resources/resource.

**Warning**  
When you upgrade your kube-hetzner module to this version, probably a lot of resources will be recreated.

To overcome this, I'll add a patch to this pull request to instead _move_ the existing resources, as described in https://developer.hashicorp.com/terraform/language/moved. However, this needs terraform version >=1.9, while the module is now at >=1.5.

aleksasiriski commented on versions.tf:
```diff
@@ -1,5 +1,5 @@
 terraform {
-  required_version = ">= 1.5.0"
+  required_version = ">= 1.9.0"
```
How does this affect OpenTofu users? Have you tested?

janfrederik replied:
Right. I forgot about opentofu.
OpenTofu is not on 1.9 yet, so the migration with the `moved` statement won't be possible. So this has to wait until opentofu catches up with v1.9.

aleksasiriski replied:
Then I suggest waiting until this is compatible with OpenTofu as well, since I've switched to it over Terraform due to license changing and I expect more people using this project may have done so as well.

janfrederik commented:
> I'm ready to merge when you are, let me know @janfrederik

Hi @mysticaltech 
I just updated with the latest code from main and fixed a null_resource that I had forgotten.  

Still waiting for the release of opentofu 1.10 with the support for moving `null_resource` to `terraform_data`. The code has already been committed to opentofu's main branch (see PR https://github.com/opentofu/opentofu/pull/2481), so now we just need a little patience for the release.

mysticaltech commented:
@janfrederik Thank you, excellent!

janfrederik commented on versions.tf:
```diff
@@ -1,5 +1,5 @@
 terraform {
-  required_version = ">= 1.8.0"
+  required_version = ">= 1.10.1"
   required_providers {
```
As stated above, opentofu 1.10.1 is needed for the automatic migration (using `moved` blocks) to work.

janfrederik commented:
I merged master into this branch again and migrated the newly added `null_resource.nat_router_await_cloud_init` to terraform_data as well.  
Ready for merge.
